YCAI/src/app.tsx

```diff
@@ -129,18 +135,25 @@ const InjectedApp: React.FC = () => {
 
 const YC_ROOT_ID = 'yc-root-injected';
 
-const ycRootNode = document.querySelector(`#${YC_ROOT_ID}`);
-if (ycRootNode === null) {
-  const ycRoot = document.createElement('div');
-  ycRoot.id = YC_ROOT_ID;
-  ycRoot.style.position = 'absolute';
-  ycRoot.style.width = "0";
-  document.body.appendChild(ycRoot);
-
-  ReactDOM.render(
-    <React.StrictMode>
-      <InjectedApp />
-    </React.StrictMode>,
-    ycRoot
-  );
-}
+const observer = new MutationObserver(() => {
+  const ycRootNode = document.querySelector(`#${YC_ROOT_ID}`);
+
+  if (ycRootNode === null) {
```

Comment:
I don't understand this observer. If I'm not mistaken the `YC_ROOT_ID` node will never be present when entering this function, and the observer will always disconnect after the first mutation so:
- I believe that `if` does nothing
- instead it should probably watch for the apparition of `YT_RELATED_SELECTOR`, then render and disconnect

Reply:
I agree the `if` is pointless, but the `YC_RELATED_SELECTOR` is already checked in `InjectedApp`, so I think can remove the check and that's it